### PR TITLE
Deploy now supports partial success and

### PR DESCRIPTION
**Pull request by erocheleau**
**Fixes issue:** #83 

Deploy now supports partial success and better error handling

### CODE REVIEW

File: src/commands/deploypackageCommand.ts [hunk 1]
```diff
@@ -95,4 +105,96 @@ export class DeploypackageCommand implements ICommand {
...
+    })
+  }
+
+  private handleDeployFailed(response: any) {
+
+    let failures = utils.asArray(response.details.componentFailures);
+    failures.forEach((fail: componentFailure) => {
+      let diag = this.printFailure(fail);
+      let uri = vscode.Uri.file("test");
+      this.diags.set(uri, [diag]); // TODO: find the file...
```
Comment (lbenie) [inline src/commands/deploypackageCommand.ts]: is the TODO still relevant ?

**Review by lbenie — COMMENTED** [review]
nnoiiceee stuff !

**Review by lbenie — APPROVED** [review]

File: src/commands/deploypackageCommand.ts [hunk 2]
```diff
@@ -15,9 +50,13 @@ export class DeploypackageCommand implements ICommand {
   // Connection handle through Salesforce
```
Comment (tedre191) [inline src/commands/deploypackageCommand.ts]: faudrait renommer la class "DeployPackageCommand" et le fichier "deployPackageCommand"

File: src/commands/deploypackageCommand.ts [hunk 3]
```diff
@@ -93,6 +103,80 @@ export class DeploypackageCommand implements ICommand {
           });
         }
       });
+  }
+
+  private handleDeployResponse(response: any) {
+    this.output.appendLine('Deploy package completed');
```
Comment (tedre191) [inline src/commands/deploypackageCommand.ts]: "Package deployed" ?

File: src/commands/deploypackageCommand.ts [hunk 4]
```diff
@@ -93,6 +103,80 @@ export class DeploypackageCommand implements ICommand {
...
+        this.displayDeployResponse(response);
+        break;
+      case DeployStatus[DeployStatus.InProgress]: // Other status
+      case DeployStatus[DeployStatus.Canceling]:
+      case DeployStatus[DeployStatus.Pending]:
+        console.log("Should not happend")
+        break;
+    }
+
+    vscode.window.showInformationMessage(`Deployment of package ${response.status}`, { title: 'Show output', action: 'SHOW_OUTPUT' })
```
Comment (tedre191) [inline src/commands/deploypackageCommand.ts]: Faudrait changer ça, c'est vraiment weird de voir "Deployement of package success" comme message

**Pull request merged**